.github/workflows/lint.yml

```diff
@@ -12,12 +12,15 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Checkout
+        uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
+          cache: 'npm'
```

Comment:
I'm curious if we are able to notice any speed up in `npm ci` steps in this or other workflows as a result of caching. With the variable GitHub Action runners it may be hard to get reliable numbers, though.

Reply:
From what I can tell from a quick small number of checks at https://github.com/open-telemetry/opentelemetry-js/actions/workflows/unit-test.yml, the run time reduced from 2m35s to about 2m with this PR.